flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -569,6 +571,12 @@ public void handleOperatorEvent(OperatorEvent event) {
             sourceReader.handleSourceEvents(((SourceEventWrapper) event).getSourceEvent());
         } else if (event instanceof NoMoreSplitsEvent) {
             sourceReader.notifyNoMoreSplits();
+        } else if (event instanceof IsBacklogEvent) {
+            eventTimeLogic.triggerPeriodicEmit(System.currentTimeMillis());
```

Comment:
We should invoke this method to send the watermark downstream in both cases. In case of event.isBacklog() == false, the downstream need to update the max watermark during backlog.

We cannot cancel the periodicEmit. Otherwise, operators that do not support backlog processing, such as multi-input operators, will not work.